packages/embed-react/src/components/make-button-component.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { CSSProperties, MutableRefObject, ReactNode, useEffect, useRef } from 'react'
+
+import { InlineStyle } from './inline-style'
+
+type ButtonComponentBaseProps = {
+  id: string
+  style?: CSSProperties
+  className?: string
+  children: ReactNode
+}
+
+type ButtonComponentProps<T> = T & ButtonComponentBaseProps
+
+type CreateFnProps<T> = Omit<ButtonComponentProps<T>, keyof ButtonComponentBaseProps>
+
+type CreateFn<T> = (id: string, props: CreateFnProps<T>) => GenericEmbed
+
+type GenericEmbed = {
+  unmount: () => void
+  open: () => void
+}
+
+function makeButtonComponent<T>(createFn: CreateFn<T>, cssFilename: string) {
+  return ({ id, children, style = {}, className = '', ...props }: ButtonComponentProps<T>) => {
+    const ref: MutableRefObject<GenericEmbed | null> = useRef(null)
```

Comment:
I'm not a fan of the union with null here, I'd consider maybe having a default object instead, not saying necessarily that it is better, probably a matter of taste

Reply:
Something like this?

```
const emptyEmbed: GenericEmbed = {
  unmount: () => {},
  open: () => {},
}
...
const ref: MutableRefObject<GenericEmbed> = useRef(emptyEmbed)
```